Doc 1734/changelog (#631)
swan-io/docs: changelog/2026/2026-04-23.mdx

---
title: 23 April 2026
type: release
---

## New features

> 📅 On **May 21, 2026**, public onboarding URLs will switch to [Onboarding v2](/preview/new-onboarding/) for all new onboardings, and the v1 public onboarding interface will be retired. The new experience delivers a cleaner onboarding flow with improved data collection for risk assessment and compliance. If you use public onboarding URLs without any prefilling, no action is needed. If you prefill onboarding fields via the API before redirecting users, refer to the [migration guide](/topics/onboarding/overview/guide-migrate/) or reach out to your Partner Integration Manager to plan your migration.

Reviewer: Hey we're not calling it v1/v2 - current/new

Reviewer: This is a very chunky paragraph, with key information buried in the prose. Suggest restructuring with key information (e.g. no action needed) clearly stated. 

Also - not everyone has a PIM so revert to our usual wording around this which is "reach out to your Swan contact" (it's something like this - verify first.

Reviewer: I'd cut the "this experience delivers xyz" - focus on the main topic.

> 📅 On **May 21, 2026**, public onboarding URLs will switch to [Onboarding v2](/preview/new-onboarding/) for all new onboardings, and the v1 public onboarding interface will be retired. The new experience delivers a cleaner onboarding flow with improved data collection for risk assessment and compliance. If you use public onboarding URLs without any prefilling, no action is needed. If you prefill onboarding fields via the API before redirecting users, refer to the [migration guide](/topics/onboarding/overview/guide-migrate/) or reach out to your Partner Integration Manager to plan your migration.

Reviewer: ```suggestion
📅 On **May 21, 2026**, public onboarding URLs will switch to [Onboarding v2](/preview/new-onboarding/) for all new onboardings, and the v1 public onboarding interface will be retired. The new experience delivers a cleaner onboarding flow with improved data collection for risk assessment and compliance. If you use public onboarding URLs without any prefilling, no action is needed. If you prefill onboarding fields through the API before redirecting to the onboarding URL, refer to the [migration guide](/topics/onboarding/overview/guide-migrate/) or reach out to your Partner Integration Manager to plan your migration.
```

## Improvements

✏️ **Custom details for the `Other` purpose**: when a supporting document with the purpose `Other` is requested, a new `purposeDetails` field on the [`SupportingDocumentPurpose` object](https://api-reference.swan.io/objects/supporting-document-purpose/) specifies what document is needed (for example, "Company structure chart"). Query `purposeDetails` under `requiredSupportingDocumentPurposes` and display the description to your account holders to guide their upload. See the [supporting documents guide](/topics/accounts/documents/#document-purposes-and-types) for details.

## API updates

### Upcoming breaking changes

> ⏱️ Starting **May 20, 2026**, Swan is reducing the [Consent Notification endpoint](/topics/users/consent/guide-configure-notifications) timeout from 3 seconds to 2 seconds. This change applies to consent flows (sensitive operations), not OAuth2 flows such as login or authentication. The shorter timeout gives Swan enough time to fall back to text message within the 3.5-second 3-D Secure (3DS) window. If your integration uses in-app consent notifications, make sure your endpoint responds within 2 seconds, ideally with some margin (around 1.5 to 1.8 seconds). If your response is slower, Swan falls back to text message. No action is required if you don't use in-app consent notifications.

Reviewer: This is quite a dense paragraph - with the endpoint messaging buried in. 

Break up the sentences particularly: 
```suggestion
⏱️ Starting **May 20, 2026**, Swan is reducing the [Consent Notification endpoint](/topics/users/consent/guide-configure-notifications) timeout from 3 seconds to 2 seconds. This change applies to consent flows (sensitive operations), not OAuth2 flows such as login or authentication. The shorter timeout gives Swan enough time to fall back to text message within the 3.5-second 3-D Secure (3DS) window. If your integration uses in-app consent notifications, make sure your endpoint responds within 2 seconds. Aim for 1.5 to 1.8 seconds to leave margin. If your response is slower, Swan falls back to text message. No action is required if you don't use in-app consent notifications.
```

> ⚠️ Starting **September 30, 2026**, [Onboarding API v1](/preview/new-onboarding/legacy-onboarding/) will be shut down and all v1 endpoints will no longer be available. [Onboarding API v2](/preview/new-onboarding/) has been generally available since March 2026, delivering improved data validation, expanded country coverage, and better support for risk assessment and compliance. If you're still using v1, migrate before this date. Refer to the [migration guide](/topics/onboarding/overview/guide-migrate/) or reach out to your Partner Integration Manager to plan your transition.

Reviewer: Is it on or starting? In the coming up section we say "On september 30, 2026". 

Align for consistency. Same goes for the May dates.

---
title: 23 April 2026
type: release
---

## New features

> 📅 On **May 21, 2026**, public onboarding URLs will switch to the [current onboarding API](/preview/new-onboarding/) for all new onboardings. The previous public onboarding interface will be retired.

Reviewer: ```suggestion
📅 On **May 21, 2026**, public onboarding URLs will switch to the [new onboarding API](/preview/new-onboarding/) for all new onboardings. The previous public onboarding interface will be retired.
```

To the new? 

current = v1
new = v2?

Is that the logic? Looking at the link its linking to new onboarding.

> If you use public onboarding URLs without any prefilling, no action is needed. If you prefill onboarding fields through the API before redirecting to the onboarding URL, refer to the [migration guide](/topics/onboarding/overview/guide-migrate/) or reach out to your Swan contact to plan your migration.

Reviewer: I don't understand why the migration guide doesn't sit with the rest of the onboarding V2 content in the Preview?
